dotcom-rendering/src/components/TableOfContents.importable.tsx

```diff
@@ -138,13 +138,28 @@ const verticalStyle = css`
 export const TableOfContents = ({ tableOfContents, format }: Props) => {
 	const [open, setOpen] = useState(tableOfContents.length < 5);
 
+	useEffect(() => {
+		const tocElement = document.querySelector(
+			'[data-component="table-of-contents"]',
+		) as HTMLElement;
+
+		const observer = new IntersectionObserver(
+			([e]) => {
+				// Automatically collapse the ToC when it becomes sticky (i.e. when it reaches the top of the viewport)
```

Comment:
Could we also add a comment that explains how the `1px` trick and why we use `boundingClientRect` a bit more? Perhaps a link to the article that explains how this 'trick' works?

https://css-tricks.com/how-to-detect-when-a-sticky-element-gets-pinned/

Reply:
I've updated the comments